### ISSUE TITLE
feat(siglip): serve the SigLIP text tower through /v1/embeddings

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1341]
## Summary

mlxcel already runs the SigLIP vision tower (`src/vision/encoders/siglip.rs`) inside several VLMs, but a standalone `siglip` checkpoint (`model_type: siglip`, `architectures: ["SiglipModel"]`) cannot be loaded and the text tower does not exist. This issue adds the text tower (token + learned position embeddings, the same pre-norm encoder block as the vision side, final LayerNorm, a projection `head`, last-position pooling) and registers `siglip` as an embedding model so `/v1/embeddings` returns the normalized text embedding used for image-text matching.

## Current behavior

- `src/models/detection.rs:124-199` has no `"siglip"` arm; a `SiglipModel` checkpoint is `Unsupported model type: siglip`.
- `src/vision/encoders/siglip.rs:147-260` implements `VisionAttention`, `VisionMLP` (with `gelu_pytorch_tanh`, line 34) and `EncoderLayer`, all private and vision-only (`VisionEmbeddings` at line 341 is a patch conv; there is no token or position-id embedding and no projection head).
- `src/vision/config.rs` `VisionConfig` has no text-side fields.

### Config

`config.json` top level: `model_type: "siglip"`, `text_config` object (all keys optional; defaults in the table), `vision_config` (ignored here).

| `text_config` key | default |
|---|---|
| `vocab_size` | 32000 |
| `hidden_size` | 768 (`D`) |
| `intermediate_size` | 3072 |
| `num_attention_heads` | 12 |
| `num_hidden_layers` | 12 |
| `max_position_embeddings` | 64 (`P`) |
| `layer_norm_eps` | 1e-6 |
| `projection_size` | `hidden_size` |
| `hidden_act` | `gelu_pytorch_tanh` |

### Tokenization

The checkpoint's `tokenizer.json` (SentencePiece unigram, normalizer lowercases) is used through `MlxcelTokenizer::encode(text, true)`, which appends `</s>` (id 1) through the post-processor. Each input is truncated to `P - 1` tokens plus the trailing `</s>`, then right-padded to exactly `P` with `pad_token_id` (1, same as `</s>`). `pad_to_max_length()` returns `Some(64)`. No attention mask is applied: every position attends to all `P` positions, which is the training-time recipe, and the pooled token is position `P - 1`.

### Forward

```
ids: [B, 64]
h = token_embedding[ids] + position_embedding[arange(64)]
for layer in 0..num_hidden_layers:
    h = h + out_proj(sdpa(q_proj(LN1(h)), k_proj(..), v_proj(..), scale = head_dim^-0.5, mask = None))
    h = h + fc2(gelu_pytorch_tanh(fc1(LN2(h))))
h = final_layer_norm(h)
pooled = head(h[:, 63, :])                     # [B, projection_size]
embedding = normalize_l2(pooled)               # pooling is fixed; 1_Pooling is not consulted
```

### Weight keys and sanitize

```
text_model.embeddings.token_embedding.weight     [32000, 768]
text_model.embeddings.position_embedding.weight  [64, 768]
text_model.encoder.layers.{i}.layer_norm1.{weight,bias}
text_model.encoder.layers.{i}.layer_norm2.{weight,bias}
text_model.encoder.layers.{i}.self_attn.{q_proj,k_proj,v_proj,out_proj}.{weight,bias}
text_model.encoder.layers.{i}.mlp.{fc1,fc2}.{weight,bias}
text_model.final_layer_norm.{weight,bias}
text_model.head.{weight,bias}                    [768, 768]
vision_model.*, logit_scale, logit_bias          (drop)
```

Sanitize: drop `vision_model.*`, `logit_*`, and any `position_ids` buffer. Keys are otherwise used as-is.

## Implementation plan

1. `src/vision/encoders/siglip.rs`: make `EncoderLayer`, `EncoderLayer::from_weights`, `VisionMLP`, `VisionAttention` `pub(crate)` and give `EncoderLayer::forward` an `Option<&MlxArray>` mask parameter (vision callers pass `None`, behavior unchanged; the sole purpose is sharing the block). Add `// Used by:` comments naming the SigLIP text tower.
2. `src/models/siglip_text.rs` (+ `siglip_text_tests.rs`): `SigLipTextArgs` (serde over `text_config` with the defaults above), `SigLipTextModel { token_embedding: UnifiedEmbedding, position_embedding: UniquePtr<MlxArray>, layers: Vec<EncoderLayer>, final_layer_norm: LayerNorm, head: UnifiedLinear, max_positions: usize }`, `from_weights(weights, args, group_size, bits)` using the `text_model.` prefix, `encode(ids) -> [B, 64, D]`, and `impl EmbeddingModel` (`embed` pools position 63, `default_pooling = LastToken` for reporting only, `embedding_dim = projection_size`, `pad_to_max_length = Some(max_positions)`).
3. `src/models/detection.rs`: `"siglip" => ModelType::SiglipText` (an embedding kind; the VLMs that embed SigLIP towers carry their own `model_type`, so this arm only fires for standalone checkpoints). `src/model_metadata.rs` entry; `src/embeddings/loader.rs` arm that reads `text_config`, loads with `load_weights_from_dir_with_subfolders`, sanitizes, converts f32 to f16 only when the checkpoint is bf16 (f32 stays f32).
4. `docs/supported-models.md` row; `docs/embeddings.md` note that inputs are lowercased by the tokenizer and capped at 64 tokens.

## Validation

(a) Unit tests in `src/models/siglip_text_tests.rs`:
- `pads_to_64_and_keeps_trailing_eos` (a 100-token input yields 63 tokens + `</s>`).
- `pooled_position_is_last` (random 1-layer model; changing token 10 of a 12-token input changes the output, changing nothing but the padding does not, since padding is the same id).
- `sanitize_drops_vision_and_logit_keys`.
- `encoder_block_shared_with_vision_is_unchanged`: the vision `EncoderLayer::forward(x, None)` output equals the pre-change output on a fixed seed (guards the refactor).

(b) Real checkpoint:
```bash
./target/release/mlxcel download google/siglip-base-patch16-224
./target/release/mlxcel embed -m models/siglip-base-patch16-224 \
  -p "a photo of a cat" -p "a photo of a kitten" -p "a photo of a cat" -p "a diagram of a car engine"
./target/release/mlxcel-server -m models/siglip-base-patch16-224 --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' -d '{"input": ["a photo of a cat", "a photo of a kitten"]}'
```
Acceptance: vectors have `projection_size` (768) entries and unit norm; cosine(cat, cat) = 1.0 within 1e-6; cosine(cat, kitten) > cosine(cat, car engine) by at least 0.1; `usage.prompt_tokens` counts real tokens (not the 64 padded slots).

## Acceptance criteria

- [ ] `src/models/siglip_text.rs` loads the text tower from a `SiglipModel` checkpoint, reusing the encoder block of `src/vision/encoders/siglip.rs`
- [ ] inputs are padded to `max_position_embeddings` and pooled at the last position, then L2-normalized
- [ ] google/siglip-base-patch16-224 passes the gate above through `mlxcel embed` and `POST /v1/embeddings`
- [ ] docs/supported-models.md updated
- [ ] detection table in src/models/detection.rs updated with a test
- [ ] cargo test --workspace --profile test-fast --features metal,accelerate passes
- [ ] cargo clippy --workspace --all-targets -- -D warnings and cargo fmt --all -- --check pass

## Out of scope

- Image embeddings through the SigLIP vision tower and attention-pooling head on `/v1/embeddings` (a follow-up once image inputs exist for VLM embedders).
- SigLIP 2 (`siglip2`) text towers.

## Dependencies

Blocked by #1353

- Part of epic #1348
